tests/BoreholeFileControllerTest.cs

```diff
@@ -0,0 +1,247 @@
+﻿using BDMS.Controllers;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Minio.Exceptions;
+using Moq;
+using System.Net;
+using System.Reactive.Linq;
+using System.Text;
+using static BDMS.Helpers;
+
+namespace BDMS;
+
+[TestClass]
+public class BoreholeFileControllerTest
+{
+    private BdmsContext context;
+    private BoreholeFileController controller;
+    private BoreholeFileUploadService boreholeFileUploadService;
+
+    private int boreholeCount;
+
+    [TestInitialize]
+    public void TestInitialize()
+    {
+        var configuration = new ConfigurationBuilder().AddJsonFile("appsettings.json").Build();
+
+        context = ContextFactory.CreateContext();
+
+        var boreholeFileUploadServiceLoggerMock = new Mock<ILogger<BoreholeFileUploadService>>(MockBehavior.Strict);
+        boreholeFileUploadServiceLoggerMock.Setup(l => l.Log(It.IsAny<LogLevel>(), It.IsAny<EventId>(), It.IsAny<It.IsAnyType>(), It.IsAny<Exception>(), (Func<It.IsAnyType, Exception, string>)It.IsAny<object>()));
+        boreholeFileUploadService = new BoreholeFileUploadService(context, configuration, boreholeFileUploadServiceLoggerMock.Object);
+
+        var boreholeFileControllerLoggerMock = new Mock<ILogger<BoreholeFileController>>(MockBehavior.Strict);
+        boreholeFileControllerLoggerMock.Setup(l => l.Log(It.IsAny<LogLevel>(), It.IsAny<EventId>(), It.IsAny<It.IsAnyType>(), It.IsAny<Exception>(), (Func<It.IsAnyType, Exception, string>)It.IsAny<object>()));
+        controller = new BoreholeFileController(context, boreholeFileControllerLoggerMock.Object, boreholeFileUploadService);
+        controller.ControllerContext.HttpContext = new DefaultHttpContext();
+
+        boreholeCount = context.Boreholes.Count();
+    }
+
+    [TestCleanup]
+    public async Task TestCleanup()
+    {
+        Assert.AreEqual(boreholeCount, context.Boreholes.Count(), "Tests need to remove boreholes they created.");
+        await context.DisposeAsync();
+    }
+
+    [TestMethod]
+    public async Task UploadAndDownload()
+    {
+        var minBoreholeId = context.Boreholes.Min(b => b.Id);
+        var content = Guid.NewGuid().ToString();
+        var firstPdfFormFile = GetFormFileByContent(content, "file_1.pdf");
+
+        // Upload
+        IActionResult response = await controller.Upload(firstPdfFormFile, minBoreholeId);
+        OkResult okResult = (OkResult)response;
+        Assert.AreEqual((int)HttpStatusCode.OK, okResult.StatusCode);
+
+        // Get all boreholeFiles of borehole
+        var boreholeFilesOfBorehole = await controller.GetAllOfBorehole(minBoreholeId);
+        Assert.IsNotNull(boreholeFilesOfBorehole.Value);
```

Comment:
REQ: Anstatt einfach zu prüfen, dass die Liste nicht `null` ist, sollte hier stattdessen mindestens der `Count` oder besser die Items in der Liste geprüft werden. Damit das geht müsste man aber nach jedem Upload die Files wieder vom Borehole abhängen und löschen. Das hätte aber den Vorteil, dass diese beiden Vorgänge (_Detach_ und _Delete_) so auch gleich implizit getestet wären.

Dieser Kommentar bezieht sich auf alle weiteren Asserts, die nur auf `null` prüfen oder auf einen simplen Count.